workers/unity/Packages/com.improbable.gdk.buildsystem/Util/BuildSupportChecker.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+public static class BuildSupportChecker
+{
+    private static readonly Dictionary<BuildTarget, string> BuildPlatformSupportModuleDirectoryNames;
+
+    static BuildSupportChecker()
+    {
+        BuildPlatformSupportModuleDirectoryNames = new Dictionary<BuildTarget, string>();
+
+        const string linuxStandaloneSupport = "LinuxStandaloneSupport";
+
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneLinux] = linuxStandaloneSupport;
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneLinux64] = linuxStandaloneSupport;
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneLinuxUniversal] = linuxStandaloneSupport;
+
+#if !UNITY_EDITOR_OSX
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneOSX] = "MacStandaloneSupport";
+#endif
+
+#if !UNITY_EDITOR_WIN
+        const string windowsStandaloneSupport = "WindowsStandaloneSupport";
+
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneWindows] = windowsStandaloneSupport;
+        BuildPlatformSupportModuleDirectoryNames[BuildTarget.StandaloneWindows64] = windowsStandaloneSupport;
+#endif
+    }
+
+    public struct BuildSupportCheckResult
+    {
+        public readonly bool CanBuild;
+        public readonly BuildTarget[] TargetsWithoutBuildSupport;
+
+        public BuildSupportCheckResult(bool canBuild, BuildTarget[] targetsWithoutBuildSupport)
+        {
+            CanBuild = canBuild;
+            TargetsWithoutBuildSupport = targetsWithoutBuildSupport;
+        }
+    }
+
+    public static BuildSupportCheckResult CheckBuildSupport(params BuildTarget[] buildTargets)
+    {
+        var editorDirectory = Directory.GetParent(EditorApplication.applicationPath);
+
+#if UNITY_EDITOR_OSX
+        var playbackEnginesDirectory = Path.Combine(editorDirectory.FullName, "PlaybackEngines");
+#else
+        var playbackEnginesDirectory = Path.Combine(editorDirectory.FullName, "Data", "PlaybackEngines");
+#endif
+
+        var buildTargetsWithoutSupport = buildTargets
+            .Where(target =>
+            {
+                if (BuildPlatformSupportModuleDirectoryNames.TryGetValue(target, out var playbackEnginesDirectoryName))
+                {
+                    return !Directory.Exists(Path.Combine(playbackEnginesDirectory, playbackEnginesDirectoryName));
+                }
+
+                // If it's not in the dictionary, then assume it's not missing support.
```

Comment:
In which case why add them to the dictionary at the top?

Reply:
It's because the dictionary is built conditionally - if a platform is not in it, it's probably one of:
- a platform that has to be installed by default (e.g. windows support for windows editor)
- a platform for which we did not implement checks for yet

Reply:
But the conditional compilation also adds the platform you are on, which means it works for the first point. For the second point we likely don't have a worker sdk that supports it if we don't check for it (otherwise we should be checking for it). If there are cases were Unity could add support for a platform that we have an sdk for but don't check for then we should print a warning for that.

Reply:
I see what you mean, I'll put the information next to other build support structures